sdk/core/System.ClientModel/src/ModelReaderWriter/ModelJsonConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A generic converter which allows <see cref="JsonSerializer"/> to be able to write and read any models that implement <see cref="IJsonModel{T}"/>.
+/// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+[RequiresUnreferencedCode("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+#endif
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+public class ModelJsonConverter : JsonConverter<IJsonModel<object>>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+{
+    /// <summary>
+    /// Gets the <see cref="ModelReaderWriterOptions"/> used to read and write models.
+    /// </summary>
+    public ModelReaderWriterOptions Options { get; }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/> with a default options of <see cref="ModelReaderWriterOptions.Json"/>.
+    /// </summary>
+    public ModelJsonConverter()
+        : this(ModelReaderWriterOptions.Json) { }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+    public ModelJsonConverter(ModelReaderWriterOptions options)
+    {
+        Options = options;
+    }
+
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return !Attribute.IsDefined(typeToConvert, typeof(JsonConverterAttribute));
+    }
+
+    /// <inheritdoc/>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public override IJsonModel<object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        using JsonDocument document = JsonDocument.ParseValue(ref reader);
+        return (IJsonModel<object>)ModelReaderWriter.Read(BinaryData.FromString(document.RootElement.GetRawText()), typeToConvert, Options)!;
```

Comment:
same here. it seems there would be a UTF8 way of doing this.

Reply:
For example, could we do:
```csharp
        IJsonModel<object> deserializer = Options.GetJsonInterface(typeToConvert); // this is the new method you are adding in the other PR
        object deserialized = deserializer.Create(ref reader, Options);
        return deserialized as IJsonModel<object>;
```

Reply:
We can definitely do this, but we will still use JsonDocument in the end.  This will at least make this class do a pass through so when the underlying implementation is updated it will pick up those changes automatically.

Reply:
I think using JsonDocument (for now is fine). The main thing that gives me a pause is the fact that this code converts UTF8 JSON (in the reader) to UTF16 (so you can pass it to FromString) and then back to UTF8 (JsonDocument.Parse converts string to UTF8)